### PR TITLE
Set dependencies

### PR DESCRIPTION
**Pull request by jr0d**

### CODE REVIEW

File: templates/kube-oidc-proxy.yaml [hunk 1]
```diff
@@ -20,7 +20,7 @@ spec:
       enabled: true
   requires:
     matchLabels:
-      kubeaddons.mesosphere.io/provides: ingresscontroller
+      kubeaddons.mesosphere.io/name: dex
```
[the remark this answers is not in the document]
Comment (shaneutt) [reply templates/kube-oidc-proxy.yaml]: :thinking:

[the remark this answers is not in the document]
Comment (jr0d) [reply templates/kube-oidc-proxy.yaml]: dex relies on traefik, but the kube-oidc-addon currently relies on the dex config.yaml to be rendered.

[the remark this answers is not in the document]
Comment (hectorj2f) [reply templates/kube-oidc-proxy.yaml]: @jongiddy mentioned https://mesosphere.slack.com/archives/CMPPXKGKX/p1570536673365100

[the remark this answers is not in the document]
Comment (hectorj2f) [reply templates/kube-oidc-proxy.yaml]: we need some clarification here :)

[the remark this answers is not in the document]
Comment (jr0d) [reply templates/kube-oidc-proxy.yaml]: Ah, we did remove the explicit dependency on config.yaml ( we were using it to set the issuerURL ). I can set it back, though there will be chatter in the logs about dex being unavailable until it comes up. The static dependency workflow was:

cert-manager -> traefik -> dex -> [kube-oidc-proxy, traefik-forward-auth, dex-k8s-authenticator]

**Comment by jr0d** [comment]
Closing, shouldn't be necessary.

**Pull request closed without merging**